src/librustdoc/html/render.rs

```diff
@@ -1369,7 +1369,9 @@ impl Context {
             self.render_redirect_pages = self.maybe_ignore_item(&item);
         }
 
-        if item.is_mod() {
+        if item.attrs.list("doc").has_word("hidden") {
+            Ok(())
+        } else if item.is_mod() {
```

Comment:
Like I said, the pass only set them as hidden but still generates the files. This change prevent the creation of the files.